test/parse.jl

```diff
@@ -1276,3 +1276,13 @@ end
 @test parse("(::A)") == Expr(Symbol("::"), :A)
 @test_throws ParseError parse("(::, 1)")
 @test_throws ParseError parse("(1, ::)")
+
+# issue #23014
+@test parse("Base.:==") == parse("Base.:(==)")
+try
+    parse(":=")
```

Comment:
Good call, thanks. Looks like `@test_throws` returns a `Base.Test.Pass` object, which has a `value` field that contains the exception.